app/src/main/java/com/mybus/activity/DisplayFavoritesActivity.java

```diff
@@ -1,11 +1,101 @@
+import android.view.View;
+import android.widget.TextView;
+
 import com.mybus.R;
+import com.mybus.adapter.FavoriteViewAdapter;
+import com.mybus.dao.FavoriteLocationDao;
+import com.mybus.listener.FavoriteDeleteListener;
+import com.mybus.listener.FavoriteEditListener;
+import com.mybus.listener.OnAddNewFavoriteListener;
+import com.mybus.listener.OnEditOldFavoriteListener;
+import com.mybus.model.FavoriteLocation;
+import com.mybus.model.SearchType;
+import com.mybus.view.FavoriteNameAlertDialog;
+
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
 
 /**
  * Created by Julian Gonzalez <jgonzalez@devspark.com>
  */
-public class DisplayFavoritesActivity extends BaseDisplayActivity {
+public class DisplayFavoritesActivity extends BaseDisplayActivity implements FavoriteEditListener, FavoriteDeleteListener, OnAddNewFavoriteListener, OnEditOldFavoriteListener {
+
+    @Bind(R.id.favorites_recycler_view)
+    RecyclerView mFavoritesRecyclerView;
+    @Bind(R.id.add_favorite_action_button)
+    FloatingActionButton mAddFavoriteActionButton;
+    @Bind(R.id.noFavorites)
+    TextView mNoFavoritesTextView;
+
+
+    public static final int EDIT_SEARCH_RESULT_ID = 1;
+    public static final int ADD_SEARCH_RESULT_ID = 2;
+
+    List<FavoriteLocation> mFavorites;
+    private int mFavoritePositionToEdit;
+
+    private RecyclerView.LayoutManager mLayoutManager;
+
+    private FavoriteViewAdapter mFavoriteViewAdapter;
+    private String newAddress;
+    private FavoriteLocation oldFavorite;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ButterKnife.bind(this);
+
+        //get all the from real
+        mFavorites = FavoriteLocationDao.getInstance(this).getAll();
+
+        mFavoritesRecyclerView.setHasFixedSize(true);
+
+        mLayoutManager = new LinearLayoutManager(this);
+        mFavoritesRecyclerView.setLayoutManager(mLayoutManager);
+
+        mFavoriteViewAdapter = new FavoriteViewAdapter(mFavorites, this, this);
+        mFavoritesRecyclerView.setAdapter(mFavoriteViewAdapter);
+
+        mAddFavoriteActionButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                DisplayFavoritesActivity.this.startSearchActivityToAddFavorite(ADD_SEARCH_RESULT_ID);
+            }
+        });
+        if (!mFavorites.isEmpty()) {
+            mNoFavoritesTextView.setVisibility(View.GONE);
+        }
+    }
+
+    private void startSearchActivityToAddFavorite(int requestCode) {
+        Intent searchIntent = new Intent(this, SearchActivity.class);
+        searchIntent.putExtra(SearchActivity.SEARCH_TYPE_EXTRA, SearchType.FAVORITE);
+        startActivityForResult(searchIntent, requestCode);
+        overridePendingTransition(0, 0);
+    }
+
+    private void startSearchActivityForFavorite(int requestCode, String favoriteAddress) {
```

Comment:
Duplicated method, doing the same thing. You should create a "startSearchActivity" method and only give the param if needed for a favorite.